### PR TITLE
Better way to do milestone post disabling - more appropriate to do it…

### PR DESCRIPTION
**Pull request by mehalshah**
… at the script level controller so it applies to all levels

### CODE REVIEW

File: dashboard/test/controllers/script_levels_controller_test.rb [hunk 1]
```diff
@@ -26,6 +26,8 @@ class ScriptLevelsControllerTest < ActionController::TestCase
     @custom_s2_l2 = create(:script_level, script: @custom_script,
                            stage: @custom_stage_2, :position => 2)
     client_state.reset
+
+    Gatekeeper.clear
```
Comment (islemaster) [inline dashboard/test/controllers/script_levels_controller_test.rb:30]: I don't know what the Gatekeeper is, or what this line does.  Will you explain this?

File: dashboard/test/controllers/script_levels_controller_test.rb [hunk 2]
```diff
@@ -832,4 +834,17 @@ def create_admin_script
...
+    assert_equal false, assigns(:view_options)[:post_milestone]
+  end
+
+  test "should not have milestone posting disabled if Milestone is not set" do
+    Gatekeeper.set('postMilestone', where: {script_name: 'Some other level'}, value: false)
+    get_show_script_level_page(@script_level)
+    assert_equal true, assigns(:view_options)[:post_milestone]
+  end
+
+
```
Comment (islemaster) [inline dashboard/test/controllers/script_levels_controller_test.rb]: Nit: extra newline.

**Comment by islemaster** [comment]
Looks good, as far as I understand it.  This is definitely better than the previous approach. :+1:

File: dashboard/test/controllers/script_levels_controller_test.rb (hunk 2 above)
Comment (mehalshah) [inline dashboard/test/controllers/script_levels_controller_test.rb]: Will fix

**Pull request merged**